src/main/java/se/diabol/jenkins/pipeline/DeliveryPipelineView.java

```diff
@@ -486,29 +488,28 @@ protected static String withoutFolderPrefix(final String projectName) {
             }
             LOG.fine("Returning: " + components);
             error = null;
-            for(int i = 0; i< components.size(); i++) {
-                components.get(i).setComponentNumber(i+1);
-            }
             return components;
         } catch (PipelineException e) {
             error = e.getMessage();
             return new ArrayList<Component>();
         }
     }
 
-    private Component getComponent(String name, AbstractProject firstJob, AbstractProject lastJob, boolean showAggregatedPipeline) throws PipelineException {
+    private Component getComponent(String name, AbstractProject firstJob, AbstractProject lastJob, boolean showAggregatedPipeline, int componentNumber) throws PipelineException {
         Pipeline pipeline = Pipeline.extractPipeline(name, firstJob, lastJob);
+        Component component = new Component(name, firstJob.getName(), firstJob.getUrl(), firstJob.isParameterized(), noOfPipelines, pagingEnabled, componentNumber);
         List<Pipeline> pipelines = new ArrayList<Pipeline>();
         if (showAggregatedPipeline) {
             pipelines.add(pipeline.createPipelineAggregated(getOwnerItemGroup(), showAggregatedChanges));
         }
         if (isFullScreenView()) {
-            pipelines.addAll(pipeline.createPipelineLatest(noOfPipelines, getOwnerItemGroup(), false));
+            pipelines.addAll(pipeline.createPipelineLatest(noOfPipelines, getOwnerItemGroup(), false, component));
```

Comment:
Why `totalNoOfPipelines` for the component cannot be set in this method instead of to be passed to `createPipelineLatest`?